### PR TITLE
Fix index.md english and word reuse

### PR DESCRIPTION
**Pull request by hugbubby**
Some terminology was wrong, and some articles/conjunctions were missing.

I'm not sure why the other commits of mine were reverted, but this one should clearly stay.

### CODE REVIEW

**Comment by l-n-s** [comment]
Why do you want to use different words? IMO, it doesn't make the message more clear, prove me wrong.

**Comment by l-n-s** [comment]
Sure, there are grammar errors in it, they need to be fixed. But this text is very important, it needs to be as simple and clear as possible. We can't swap words for every PR. :)
See my notes.

**Comment by hugbubby** [comment]
I don't think you understand - this is just the semantically/gramatically correct version of what's being said, in English. It's not just swapping word choice - full-featured isn't a word in the English language.

**Comment by l-n-s** [comment]
I understand, you are absolutely right. Let me try to explain what I mean here.

I2P terminology is a mess, "I2P" means a lot of things.
I2P is a protocol, I2P is an actual public network, I2P is a software written in Java, I2P is a community project, which develops the software and maintains the network. We're trying to make it a little better in the i2pd's documentation.

We say here, that i2pd is an I2P client implementation. And later in the text, we explain what I2P client is. If we say it's a protocol implementation, we'll need to rewrite the other part, so it can make sense. I'm not sure it will make the message more clear though.

It is also not obvious to me, why have you changed the word "software" to "application" and the word "using" to "traversing"?

Other than that, the patch looks great, thanks for you contribution! :+1:

**Comment by l-n-s** [comment]
No reply? OK then, I'm going to commit grammar fixes by myself.

**Comment by hugbubby** [comment]
In English, software doesn't require the article "a", and the closest change I could think of to make the sentence work was to remove "software" and add "application". Traversing is just a more descriptive word.

We can change the header back and instead write "client for the I2P protocol" or "client for the invisible internet protocol", I just wanted to make it clear that I2P was a protocol and i2pd is software for accessing the i2p network.

File: docs/index.md [hunk 1]
```diff
@@ -3,14 +3,14 @@
 Invisible Internet Protocol (daemon)
 ====================================
 
-i2pd (I2P Daemon) is a full-featured C++ implementation of I2P client.
+i2pd (I2P Daemon) is a fully-featured C++ implementation of the I2P protocol.
```
Comment (xanoni) [inline docs/index.md:6]: What about:

`i2pd (I2P Daemon) is a fully-featured I2P client implementation written in C++.`

File: docs/index.md [hunk 2]
```diff
@@ -3,14 +3,14 @@
...
 
 I2P (Invisible Internet Protocol) is a universal anonymous network layer. 
-All communications over I2P are anonymous and end-to-end encrypted, participants
+All communications over I2P are anonymous and end-to-end encrypted, and participants
```
Comment (xanoni) [inline docs/index.md:9]: OK sure

File: docs/index.md [hunk 3]
```diff
@@ -3,14 +3,14 @@
...
 don't reveal their real IP addresses. 
 
-I2P client is a software used for building and using anonymous I2P 
-networks. Such networks are commonly used for anonymous peer-to-peer 
+An I2P client is an application for building and traversing anonymous I2P
```
Comment (xanoni) [inline docs/index.md:12]: What about:

```
An I2P client is used to build and access anonymous I2P networks.
Such networks are commonly used for both anonymous peer-to-peer
applications (e.g., filesharing, cryptocurrencies) and anonymous client-
server applications (e.g., websites, instant messengers, chat-servers).
```

**Review by xanoni — CHANGES_REQUESTED** [review]
Lol maybe we should resolve this after almost 3 years. Tried to incorporate people's feedback.

**Pull request closed without merging**